onnxruntime/core/optimizer/graph_transformer_mgr.cc

```diff
@@ -2,25 +2,58 @@
 // Licensed under the MIT License.
 
 #include "core/optimizer/graph_transformer_mgr.h"
+#include "core/optimizer/graph_transformer_factory.h"
+
 using namespace onnxruntime;
 using namespace ::onnxruntime::common;
 
 namespace onnxruntime {
 
-Status GraphTransformerManager::ApplyAll(Graph& graph) const {
+Status GraphTransformerManager::ApplyTransformations(Graph& graph, const std::vector<TransformerLevel>& levels) const {  
   for (unsigned step = 0; step < steps_; ++step) {
     bool changed = false;
-    for (auto& transformer : transformers_) {
-      bool t_changed = false;
-      Status s = transformer->Apply(graph, t_changed);
-      if (!s.IsOK()) {
-        return s;
+    // apply all transformations registered for every enabled level in the levels list
+    for (const auto& level : levels) {
+      if (!(optimizations_enabled_ & level)) {
+        continue;
       }
-      changed = changed || t_changed;
+
+      const auto& transformers = factory_->GetTransformers(level);
+      ORT_ENFORCE(Apply(graph, transformers, changed).IsOK());
     }
```

Comment:
Don't you need this? Otherwise 'changed' is just the result of the last transform, and not an overall status for the loop.